src/scaling.jl

```diff
@@ -12,93 +12,65 @@ Represents the no-op scaling which simply returns the `data` it is applied on.
-* `std::NamedTuple`: similar to `mean` but for standard deviation values.
 
 # Keyword arguments to `apply`
 * `inverse=true`: inverts the scaling (e.g. to reconstruct the unscaled data)
 * `eps=1e-3`: replaces all 0 values in `std` before scaling (if `inverse=false`)
 """
 struct MeanStdScaling <: AbstractScaling
-    mean::NamedTuple
-    std::NamedTuple
-end
-
-"""
-    MeanStdScaling(data; kwargs...) <: Scaling
-
-Construct a [`MeanStdScaling`](@ref) using the mean and standard deviation of the data.
-
-!!! note
-    `dims` and `cols` keyword arguments must be specified the same way when constructing
-    and applying the transform.
-    Otherwise, the results will be inconsistent, or an error may occur.
-
-# Keyword arguments
-* `dims=:`: for `AbstractArray` data, the dimension(s) to compute statistics along.
-* `cols=nothing`: for `Table` data, the column names to compute statistics for.
-"""
-function MeanStdScaling(data; kwargs...)
-    μ, σ = compute_stats(data; kwargs...)
-    return MeanStdScaling(μ, σ)
-end
+    μ::Real
+    σ::Real
+
+    """
+        MeanStdScaling(A::AbstractArray; dims=:, inds=:) -> MeanStdScaling
+        MeanStdScaling(table, cols=nothing) -> MeanStdScaling
+
+    Construct a [`MeanStdScaling`](@ref) transform from the statistics of the given data.
+    Default behaviour is to consider all data when computing the mean and standard deviation.
+    This can be restricted to certain slices via the keyword arguments (see below).
+
+    # `AbstractArray` keyword arguments
+    * `dims=:`: the dimension along which to take the `inds` slices.
+    * `inds=:`: the slices of the array to use in computing the statistics. Default uses all slices.
+
+    # `Table` keyword arguments
+    * `cols=nothing`: the columns to use in computing the statistics. Default uses all columns.
+
+    !!! note
+        The keyword arguments must be specified the same way when you construct and [`apply`](@ref)
+        the transform, else the results will be inconsistent, or throw an error.
+    """
+    function MeanStdScaling(A::AbstractArray; dims=:, inds=:)
+        dims == Colon() && return new(compute_stats(A)...)
+        return new(compute_stats(selectdim(A, dims, inds))...)
+    end
 
-function compute_stats(A::AbstractArray; dims=:)
-    if dims == Colon()
-        μ = (all = mean(A), )
-        σ = (all = std(A), )
-    else
-        μ_pairs = [(Symbol(i), x) for (i, x) in enumerate(mean(A; dims=dims))]
-        σ_pairs = [(Symbol(i), x) for (i, x) in enumerate(std(A; dims=dims))]
+    function MeanStdScaling(table; cols=nothing)
+        Tables.istable(table) || throw(MethodError(MeanStdScaling, table))
+        columntable = Tables.columns(table)
 
-        μ = (; μ_pairs...)
-        σ = (; σ_pairs...)
+        cols = _to_vec(cols)  # handle single column name
+        cnames = cols === nothing ? propertynames(columntable) : cols
+        data = reduce(vcat, [getproperty(columntable, c) for c in cnames])
```

Comment:
This is probably just because I'm set in the old way, but I'm concerned that the meaning of `cols` is confusing. I expect it to independently normalise the `cols` (which I think is a common use-case and a downside of this design). Instead it computes the stats over the whole block.

This behaviour is stated in the docstring but I think it's worth emphasising.

Reply:
is there somewhere else you'd like to see it emphasised?

Reply:
Just the docstring is fine